telemetry/statedata.go

```diff
@@ -0,0 +1,30 @@
+package telemetry
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/hashicorp/go-metrics"
+)
+
+// StateDataCounter increments a counter with the given data type and success status.
+func StateDataCounter(ctx context.Context, dataType string, isSuccessful func() bool) {
+	success := "false"
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	blockHeight := sdkCtx.BlockHeight()
+	if isSuccessful() {
+		success = "true"
+	}
+
+	telemetry.IncrCounterWithLabels(
+		[]string{"state_data"},
```

Comment:
1. Make sure to rename this too if we rename the function.
2. Consider moving it to a local constant